src/logics/morphoblue/logic.withdraw.ts

```diff
@@ -0,0 +1,57 @@
+import { Service } from './service';
+import * as common from '@protocolink/common';
+import * as core from '@protocolink/core';
+import { getMarket, getMarkets, supportedChainIds } from './configs';
+
+export type WithdrawLogicTokenList = Record<string, common.Token[]>;
+
+export type WithdrawLogicFields = core.TokenOutFields<{ marketId: string }>;
+
+export type WithdrawLogicOptions = Pick<core.GlobalOptions, 'account'>;
+
+export class WithdrawLogic extends core.Logic implements core.LogicTokenListInterface, core.LogicBuilderInterface {
+  static id = 'withdraw';
+  static protocolId = 'morphoblue';
+  static readonly supportedChainIds = supportedChainIds;
+
+  async getTokenList() {
+    const tokenList: WithdrawLogicTokenList = {};
+    const service = new Service(this.chainId, this.provider);
+
+    // TODO: get all the markets vs whitelisted markets
+    const markets = getMarkets(this.chainId);
+
+    for (const market of markets) {
+      const loanTokens = await service.getLoanTokens(market.id);
+      for (const loanToken of loanTokens!) {
+        tokenList[market.id] = [];
+        if (loanToken.isWrapped) tokenList[market.id].push(loanToken.unwrapped);
+        tokenList[market.id].push(loanToken);
+      }
+    }
+
+    return tokenList;
+  }
+
+  async build(fields: WithdrawLogicFields, options: WithdrawLogicOptions) {
```

Comment:
Considering that morpho accrues interests before execution, do we need to handle the condition of withdrawAll, similar to the repayAll?

Reply:
I think it would be cool to have this option for users. What does @zodahu think about this?

Reply:
Is this similar to Aave's atoken? If so, I think we can follow Aave's approach and not support `withdrawAll`, meaning that a small amount of dust collateral is acceptable. This is because repayAll solves that a little bit of debt dust  cause errors in withdrawing ~all collateral.

Reply:
This `withdraw()` is for loan tokens, not collateral tokens. It is more like withdrawing Compound v3 base token without `cToken`. It seems that Morpho does not require users to have 0 supplied position before doing a borrow. I think we can leave the dust since it won't cause any unexpected errors.

Reply:
We can update in the future if there is such use case anyway. Let's keep the current design for now.